llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -41273,6 +41273,33 @@ static SDValue combineTargetShuffle(SDValue N, const SDLoc &DL,
 
     return SDValue();
   }
+  case X86ISD::VPERMV3: {
+    // VPERM[I,T]2[B,W] are 3 uops on Skylake and Icelake so we try to use
+    // VPERMV.
+    if (VT.is512BitVector() || (VT.is256BitVector() && !Subtarget.hasEVEX512()))
+      return SDValue();
```

Comment:
Not sure, but it might be better to check the NVT type instead of VT?

Reply:
The drawback is we need to construct NVT before we know if it's necessary. Let me know if you still think checking NVT is better.

Reply:
MVTs are cheap to construct - I'd say go for it.

Reply:
Done.